frontend/pages/hosts/ManageHostsPage/components/DeleteSecretModal/DeleteSecretModal.tsx

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import Modal from "components/Modal";
+import Button from "components/buttons/Button";
+import { ITeam } from "interfaces/team";
+import { IEnrollSecret } from "interfaces/enroll_secret";
+
+interface IDeleteSecretModal {
+  selectedTeam: number;
+  teams: ITeam[];
+  onDeleteSecret: () => void;
+  toggleDeleteSecretModal: () => void;
+}
+
+interface IRootState {
+  app: {
+    enrollSecret: IEnrollSecret[];
+  };
+}
+
+const baseClass = "delete-secret-modal";
+
+const DeleteSecretModal = ({
+  selectedTeam,
+  teams,
+  onDeleteSecret,
+  toggleDeleteSecretModal,
+}: IDeleteSecretModal): JSX.Element => {
+  const globalSecret = useSelector(
+    (state: IRootState) => state.app.enrollSecret
+  );
+
+  const renderTeam = () => {
+    if (typeof selectedTeam === "string") {
+      selectedTeam = parseInt(selectedTeam, 10);
+    }
+
+    if (selectedTeam === 0) {
+      return { name: "No team", secrets: globalSecret };
```

Comment:
```suggestion
      return { name: "No team" };
```

You could also just refactor this to return the team name directly as a string rather than the team object